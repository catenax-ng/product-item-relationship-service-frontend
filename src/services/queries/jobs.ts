import { useMutation } from "@tanstack/react-query";
import { useQuery } from "../../lib/react-query";
import { IRSRequestBody } from "../../types/jobs";
import { getCurrentEnvironment } from "../../utils/sessionStorageHandling";
import { selectAPI } from "../api/jobs.api";

export const useFetchJobById = ({ id }: { id: string }, refetchInterval: false | number = false) => {
  const serverEnv = getCurrentEnvironment();
  const jobAPI = selectAPI(serverEnv);
  return useQuery(["jobs", serverEnv, id], () => jobAPI.fetchJobById(id), {
    refetchInterval,
  });
};

export const useFetchJobs = (page: number, refetchInterval: false | number = false) => {
  const serverEnv = getCurrentEnvironment();
  const jobAPI = selectAPI(serverEnv);
  return useQuery(["jobs", serverEnv, page], () => jobAPI.fetchJobs(page), { refetchInterval });
};

export const useCancelJobs = () => {
  const serverEnv = getCurrentEnvironment();
  const jobAPI = selectAPI(serverEnv);
  return useMutation(jobAPI.cancelJob);
};

export const useCreateJob = () => {
  const serverEnv = getCurrentEnvironment();
  const jobAPI = selectAPI(serverEnv);
  return useMutation({
    mutationFn: (data: IRSRequestBody) => {
      return jobAPI.createJob(data);
    },
  });
};
