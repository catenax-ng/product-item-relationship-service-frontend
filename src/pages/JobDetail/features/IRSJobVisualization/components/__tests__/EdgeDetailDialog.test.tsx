import { JobsDemoDataSuccess } from "../../../../../../services/api/__test__/jobs.exampleData";
import { fireEvent, render, screen } from "../../../../../../testing/test-utils";
import { EdgeDetailDialog } from "../EdgeDetailDialog";

it("does not open the dialog", () => {
  const { container } = render(<EdgeDetailDialog edge={undefined} onClose={jest.fn} />);
  expect(container).toMatchSnapshot();
});

it("opens the edge detail dialog", () => {
  const edge = JobsDemoDataSuccess.relationships[0];
  const closeFn = jest.fn();
  render(<EdgeDetailDialog edge={edge} onClose={closeFn} />);
  const button = screen.getByRole("button", {
    name: /close/i,
  });
  fireEvent.click(button);
  expect(closeFn).toBeCalledTimes(1);
});
