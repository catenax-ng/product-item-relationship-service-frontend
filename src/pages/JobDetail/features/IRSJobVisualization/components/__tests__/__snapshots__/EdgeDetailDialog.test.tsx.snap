// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not open the dialog 1`] = `<div />`;

exports[`opens the edge detail dialog 1`] = `
<div
  aria-hidden="true"
/>
`;
